libs/blocks/table/table.js

```diff
@@ -0,0 +1,172 @@
+import {decorateBlockText, decorateButtons, getBlockSize} from '../../utils/decorate.js';
+
+export default function init(el) {
+  const rows = Array.from(el.children);
+  rows.forEach((row, rdx) => {
+    row.classList.add(`row`, `row-${rdx + 1}`);
+    const cols = Array.from(row.children);
+    cols.forEach((col, cdx) => {
+      col.classList.add(`col`, `col-${cdx + 1}`);
+      if (rdx === 1) {
+        col.classList.add('col-heading');
+      }
+    });
+  });
+
+  handleSectionHead(el);
+  handleHeading(el);
+  handleHovering(el);
+  handleHighlight(el);
+  handleScrollEffect(el);
+
+  applyStylesBasedOnScreenSize(el);
+  window.addEventListener('resize', () => applyStylesBasedOnScreenSize(el));
```

Comment:
It's recommended to use `matchMedia` instead of listening to the window resize event.   Also if using this event it should probably be debounced